Gemfile

```diff
@@ -2,3 +2,11 @@ source 'https://rubygems.org'
 
 # Specify your gem's dependencies in kafo_parsers.gemspec
 gemspec
+
+puppet_version = ENV['PUPPET_VERSION'] || ENV['puppet']
+puppet_spec = puppet_version ? "~> #{puppet_version}" : '< 5.0.0'
+gem 'puppet', puppet_spec
+
+if puppet_version.nil? || puppet_version >= '3.7'
```

Comment:
I was trying to keep the logic inside the Gemfile for handling multiple Puppet versions, so it's easier for a developer to install a certain Puppet version and get the right extras (e.g. puppet-strings) and to avoid editing Gemfiles from Jenkins jobs. This should make it easier for a developer to reproduce a Jenkins build on a particular version.

The `puppet` environment variable came from the current name of the Jenkins job matrix axis, while `PUPPET_VERSION` came from our Puppet module tests where this pattern's used. I could remove `puppet` here and rename the axis in Jenkins after it's merged.